### PR TITLE
Add configuration file option for SQLFluff

### LINKED ISSUE #1199: Allow SQLFluff configuration to be read from usual location
**Is your feature request related to a problem? Please describe.**
SQLFluff 0.9.0 added the ability to specific a config file location in the command line, rather than picking it up from the directory tree being linted.

**Describe the solution you'd like**
Add `.sqlfluff` file to TEMPLATES directory and necessary code changes to pick it up (`sqlfluff lint --config CONFIG_FILE)`

The only caveat is that any command-line supplied config will override any other config, so it should be commented out by default so anyone using a local one in their code base can continue to use this.

**Describe alternatives you've considered**
Could leave it as is, but would be nice for consistency to be able to do this.

**Additional context**
I recently opened a PR to add this to the GitHub super linter: https://github.com/github/super-linter/pull/2361
I presume similar changes are needed here as this started life as a fork of that.

### PR DESCRIPTION
**Pull request by tunetheweb**
Fixes #1199 by allowing a `.sqfluff` config file to be picked up from the usual location (`.github/linters` I presume?).

## Proposed Changes

1. Add default `.sqlfluff` file to TEMPLATES folder - commented out
2. Add config to use that `.sqlfluff` file

Note this is commented out as any config files explicitly used by the command line supersedes any local one and people may prefer to use a local one. In effect the empty config file will be ignored, unless uncommented it. If no other config file is provided, then it will just use the defaults.

First contribution here and don't know the Mega linter so let me know if I missed anything. Also unsure how to test this?

I'm one of the SQLFluff maintainers btw so feel free to ask me any questions about that.

### Author/Contributor
- [x] If documentation is needed for this change, has that been included in this pull request

### Reviewing Maintainer
- [ ] Label as `breaking` if this is a large fundamental change
- [ ] Label as either `automation`, `bug`, `documentation`, `enhancement`, `infrastructure`, or `performance`

### CODE REVIEW

File: megalinter/descriptors/sql.megalinter-descriptor.yml [hunk 1]
```diff
@@ -26,19 +26,20 @@ linters:
   - linter_name: sqlfluff
     linter_url: https://www.sqlfluff.com/
     linter_repo: https://github.com/sqlfluff/sqlfluff
-    linter_banner_image_url: https://raw.githubusercontent.com/sqlfluff/sqlfluff/master/images/sqlfluff-wide.png
+    linter_banner_image_url: https://raw.githubusercontent.com/sqlfluff/sqlfluff/main/images/sqlfluff-wide.png
     linter_image_url: https://docs.sqlfluff.com/en/stable/_static/images/sqlfluff-lrg.png
     linter_rules_url: https://docs.sqlfluff.com/en/stable/rules.html
     linter_rules_configuration_url: https://docs.sqlfluff.com/en/stable/configuration.html
+    config_file_name: .sqlfluff
     config_file_name: setup.cfg
```
Comment (nvuillam) [inline megalinter/descriptors/sql.megalinter-descriptor.yml:34]: @tunetheweb only one config_file_name, you may want to remove the one with setup.cfg :)

**Review by nvuillam — CHANGES_REQUESTED** [review]
plz check comment :)

File: megalinter/descriptors/sql.megalinter-descriptor.yml (hunk 1 above)
Comment (tunetheweb) [reply megalinter/descriptors/sql.megalinter-descriptor.yml:34]:
```suggestion
```

**Review by nvuillam — APPROVED** [review]

**Comment by nvuillam** [comment]
@tunetheweb ouch... it seems to be an internal sqlfluff error, do you have an idea about the cause ?

image

**Comment by tunetheweb** [comment]
Ohh this is a bug in our code. I can reproduce locally. Looking...

**Comment by nvuillam** [comment]
@tunetheweb I have the same error out of this PR, it blocks new MegaLinter releases, too bad coz the Gitlab Merge Requests reporter is finally ready and except if I downgrade SQLFluff I can't release it 🤣 

When can we expect a sqlfluff patch with the bug fix ? :)

**Comment by tunetheweb** [comment]
Rule L026 (which was given a rewrite recently) checks columns in the SELECT clause exist in the FROM clause. Looks like it doesn't handle DELETE statements which have no SELECT clause :-(

It's a simple fix, but we literally just released a version today. Will see if I can persuade them to do another...

**Comment by nvuillam** [comment]
@tunetheweb it's a version with a blocking bug for probably lots of users, it could be a reason to release a patch with the hotfix ^^

Please let me know, if it takes too long I'll just downgrade and cancel the downgrade later :)

**Comment by tunetheweb** [comment]
OK the fix is merged, and likely do another release tomorrow. Going to give it one more day to see if we've introduced any other new bugs, plus we've another one we're working on.

So you can either downgrade (this PR can still be merged with 0.9.0 or above) or if no hurry then wait for that release. I'll keep you posted.

**Comment by nvuillam** [comment]
I downgraded and updated your branch, CI will probably say it's ok now :)

**Comment by tunetheweb** [comment]
FYI, 0.9.3 was just released with this fix (and an all new test suite to make sure it never happens again 😁)

**Comment by nvuillam** [comment]
@tunetheweb thanks for the info, I'll remove the downgrade ^^

**Pull request merged**